### PR TITLE
Development (Work in Progress)

### PR DESCRIPTION
**Pull request by rudolfnep**
- Added License file
- Updated package.json with a new license
- Updated the package `openzeppelin-solidity`
- Fixed a bug in `TreasuryUpdated` in the constructor
- Updated the function `getInfo` and dropped `totalNepPaired` and `maxStake`
- Updated `Pair struct` to include `pancakePair` address
- Updated `addOrUpdatePair` to include `pancakePair` address
- Testing `_releaseBondToSender` feature
- Added a new interface `IPancakePairLike` to enable approvals
- Fixed a bug in `releaseBond` by allowing PancakeRouter to transfer liquidity before calling the function `removeLiquidity`
- Dropped `nepStaked` and `bondTokenStaked` from `BondReleased` event
- Added a library NTransferUtilV1 which ensures token transfers are not malicious
- Refactored BEP-20 transfers to use `NTransferUtilV1`.
- Dropped `path` array from `Pair` struct
- Refactored the function `getNepRequired` to use PancakeSwap reserve balance for accuracy
- Refactored the variable `nepAmount` to `nepDesired` in `_addLiquidity` function for brevity
- Fixed a bug in `_addLiquidity` function which resulted in wrong amount of `_totalNepPaired` being stored
- Added `pause` and `unpause` features
- Dropped unused features of `IPancakePairLike`
- Added a fake `MaliciousToken` for testing

### CODE REVIEW

File: contracts/IPancakePairLike.sol [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: BUSL-1.1
```
Comment (nmtokyo) [inline contracts/IPancakePairLike.sol:1]: Consider dropping unused functions.

**Comment by nmtokyo** [comment]
You must ensure the amount transfer call actually results in the exact amount being transferred. Example:

```solidity
// SPDX-License-Identifier: BUSL-1.1
pragma solidity >=0.4.22 <0.9.0;
import "openzeppelin-solidity/contracts/utils/math/SafeMath.sol";
import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";

contract NUtils {
  using SafeMath for uint256;

  function safeTransfer(
    IERC20 token,
    address recipient,
    uint256 amount
  ) public {
    uint256 pre = token.balanceOf(recipient);
    token.transfer(recipient, amount);
    uint256 post = token.balanceOf(recipient);

    require(post.sub(pre) == amount, "Invalid transfer");
  }

  function safeTransferFrom(
    IERC20 token,
    address sender,
    address recipient,
    uint256 amount
  ) public {
    uint256 pre = token.balanceOf(recipient);
    token.transferFrom(sender, recipient, amount);
    uint256 post = token.balanceOf(recipient);

    require(post.sub(pre) == amount, "Invalid transfer");
  }
}
```

**Comment by rudolfnep** [comment]
Ready for another review.

**Comment by nmtokyo** [comment]
Looks clean.

**Review by nmtokyo — APPROVED** [review]

**Comment by rudolfnep** [comment]
Updated the PR description. Ready for merge.

**Pull request merged**